optparse.php

```diff
@@ -766,6 +766,10 @@ private function _process_option(&$option, &$rargs,
         // If only one value, set it directly as the value (not in an array)
         if ( $option->nargs == 1 ) {
             $value = $value[0];
+            // Treat the option as the same one until another dash (-) is found to specify we're looking at a new one.
+            while (is_array($rargs) && isset($rargs[0]) && is_string($rargs[0]) && $rargs[0][0] !== '-') {
+                $value .= ' '.array_shift($rargs);
+            }
```

Comment:
If I understand correctly what this part is doing, it is looking pretty weird to me since it's making the library behave in a way that no other unix/linux program would interpret argument values. usually if one wants to have multiple words as the value of a single option, it would be customary to quote all of the words so that they come up as the one value for the argument.

e.g.:

~~~
some_cli_tool --option "this is all one value for the argument"
~~~